pkg/tools/callback/readme.md

```diff
@@ -0,0 +1,51 @@
+## Why?
+
+NSM need a server to call client via GRPC, for this purpuse in native way it required to 
+have client serve a GRPC server on unix socket or TCP. 
+But to simplify communication between client and server a better way is to have 
+only one connection between client and server.
+
+And this grpc.callback library is intended to perform required functionality using 
+grpc bi-directional streams.
+
+## How?
+
+A Server declared as usual except:
+
+```go
+server := grpc.NewServer()
+...
+// register normal APIs on server
+...
+// Register callback GRPC endpoint and callback.Server will hold callback clients to be callable.
+callbackServer := callback.NewServer(callback.IdentityByAuthority)
+//
+callback.RegisterCallbackServiceServer(server, callbackServer)
+...
+_ = server.Serve(listener)
+```
+
+And then require to call client.
+```go
+nsmClientGRPC, _ := callbackServer.NewClient(context.Background(), clientID)
```

Comment:
Its looks like what we have here is a callbackServer.NewClient(...) that is returning a grpc.ClientConnInterface

Given that, might it not make more sense to mirror more closely what we already have API wise from grpc?

perhaps something like
```golang
cc := callbackServer.DialContex(ctx,target)
nsmClient := networkservice.NetworkServiceClient(cc)
```

Honestly I'd prefer if we could figure it out to have something like:
```golang
cc := callback.DialContex(ctx,target,WithCallbackServer(...))
nsmClient := networkservice.NetworkServiceClient(cc)
```
Where target can be in some form of a callback URL that indicates we are doing callback, WithCallBackServer(...) provides whatever parameters we need to start the CallBackServer if its not present... and if the target is *not* a callback URL we just fall through to the normal grpc.DialContext(...)

This approach would then allow folks to take advantage of callback if and only if they got a callback URL, and otherwise follow more or less the same pattern they use now for GRPC clients.  That said... all of the details of how to do that are not yet clear in my head.

Reply:
Sounds good, will check and update PR.